src/main/java/de/codecentric/centerdevice/util/StageUtils.java

```diff
@@ -46,12 +56,49 @@ public static void focusNextStage() {
 		}
 	}
 
+	@SuppressWarnings("unchecked")
 	public static ObservableList<Stage> getStages() {
-		return StageHelper.getStages();
+		if (stages == null) {
+			// Java 9
+			try {
+				windows = (ObservableList<Window>)Window.class.getMethod("getWindows").invoke(null);
+				stages = FXCollections.observableArrayList();
+				windows.addListener(new ListChangeListener<Window>() {
+					@Override
+					public void onChanged(javafx.collections.ListChangeListener.Change<? extends Window> c) {
+						updateStages();
```

Comment:
I think it creates a memory leak as it will prevent closed (and potentially no longer used) stages from being GCed as we keep a global reference .